### PR TITLE
2490 unsafely replace dashes with underscores in prometheus metrics

### PR DESCRIPTION
**Pull request by akaspin**

### CODE REVIEW

File: metrics-prom_test.go [hunk 1]
```diff
@@ -91,7 +91,7 @@ app_t1_stddev{service="test"} 0
 	t.Run("counter", func(t *testing.T) {
 		r := metrics.NewRegistry()
 		metrics.GetOrRegisterCounter("app no_label", r).Inc(2)
-		metrics.GetOrRegisterCounter("app,l1=1 with_label", r).Inc(4)
+		metrics.GetOrRegisterCounter("app,l1=1 with-label", r).Inc(4)
 		metrics.GetOrRegisterCounter("app,l1=2 with_label", r).Inc(5)
```
Comment (gburanov) [inline metrics-prom_test.go]: why not here as well?

File: metrics-prom.go [hunk 2]
```diff
@@ -238,7 +238,7 @@ func (p *PrometheusMetrics) extractSignature(raw string) (name, labels string, e
 	}
 	name = split[1]
 	split = strings.Split(split[0], ",")
-	name = split[0] + "_" + name
+	name = strings.Replace(split[0] + "_" + name, "-", "_", -1)
```
Comment (gburanov) [inline metrics-prom.go]: maybe lets create a function for this (just for documenation) - like normalizeNameForPrometheus

File: metrics-prom_test.go (hunk 1 above)
Comment (akaspin) [reply metrics-prom_test.go]: To check they'll exported as one counter.

File: metrics-prom.go (hunk 2 above)
Comment (akaspin) [reply metrics-prom.go]: This was first idea. But seems useless because we can inspect our code and errors are reported. Damn Sarama uses hard dependency.

**Comment by gburanov** [comment]
I think this one is better

**Pull request merged**